### PR TITLE
Portabled: fix inspect on image attached as directory

### LINKED ISSUE #14634: portablectl inspect fails with "lacks os-release data" when the image attached is a directory
**systemd version the issue has been seen with**

Both v239 and latest master (commit id fe888c79f50b)

**Used distribution**

Debian stable

**Expected behaviour you didn't see**

portablectl inspect successfully returns information about the image

**Unexpected behaviour you saw**

portablectl inspect fails with "lacks os-release data" error, despite os-release being there. Inspect on the exact same image wrapped in a squashfs and attached works fine.

**Steps to reproduce the problem**

Minimal repro steps:

```
$ mkdir -p squashfs/usr/lib/systemd/system squashfs/etc
$ cp /etc/os-release squashfs/usr/lib/
$ ln -sr squashfs/usr/lib/os-release squashfs/etc/
$ cat <<EOF>squashfs/usr/lib/systemd/system/test1.service
[Service]
Type=oneshot
ExecStart=echo hello one
EOF
$ cat <<EOF>squashfs/usr/lib/systemd/system/test2.service
[Service]
Type=oneshot
ExecStart=echo hello two
EOF
$ mksquashfs squashfs/ test.raw -all-root
$ sudo portablectl attach $PWD/squashfs/ test1.service
$ sudo portablectl attach $PWD/test.raw test2.service
```

Result:

```
$ portablectl list
NAME     TYPE      RO CRTIME MTIME                       USAGE STATE   
squashfs directory no n/a    n/a                         n/a   attached
test     raw       no n/a    Wed 2020-01-22 12:27:19 GMT 4.0K  attached

2 images listed.
$ portablectl inspect squashfs test1.service
(Matching unit files with prefixes 'test1.service'.)
Failed to inspect image metadata: Image '/etc/portables/squashfs' lacks os-release data, refusing.
$ portablectl inspect test test2.service
(Matching unit files with prefixes 'test2.service'.)
Image:
        /etc/portables/test.raw
Portable Service:
        n/a
Operating System:
        Debian GNU/Linux 10 (buster)
Unit files:
        test2.service
```

The error printed by portabled with debug level is: ```Couldn't acquire os-release file, ignoring: Not a directory```

Running it under gdb, it's the openat called in chase_symlinks in src/basic/fs-util.c that fails and returns "not a directory":

https://github.com/systemd/systemd/blob/fe888c79f50b6f40bb1918301daa2c4d87dcd7bf/src/basic/fs-util.c#L905

(with chase_symlinks being called by portabled via open_os_release: https://github.com/systemd/systemd/blob/fe888c79f50b6f40bb1918301daa2c4d87dcd7bf/src/shared/os-util.c#L39 )

### PR DESCRIPTION
**Pull request by bluca**
When a portable image is attached as a directory, inspect will fail
as the directory will be symlinked under [/run|/etc]/portables and
chase_symlinks will try to open the resolved link to the full
absolute path of the image under [/run|/etc]/portables/<name>.

(eg: /run/portables/test/home/user/test/etc/os-release)

Resolve the symlink beforehand to fix the issue.

Fixes #14634

### CODE REVIEW

**Comment by poettering** [comment]
Hmm, this doesn't look right. The first thing extract_now() does is invoke open_os_release() which internally already uses chase_symlinks() with the root parameter initialized. Hence this shouldn't be necessary to chase beforehand?

**Comment by bluca** [comment]
> Hmm, this doesn't look right. The first thing extract_now() does is invoke open_os_release() which internally already uses chase_symlinks() with the root parameter initialized. Hence this shouldn't be necessary to chase beforehand?

Yep, but that doesn't work currently unfortunately - see the linked issue: https://github.com/systemd/systemd/issues/14634 It might be possible to fix the logic in chase_symlinks, but as mentioned on the issue it's used in hundreds of places so I'm a bit wary of meddling with it.

**Comment by bluca** [comment]
TL;DR when a directory is attached, it's symlinked under /etc/portables, and the logic in chase_symlinks at some point tries to open /path/to/directory/etc/os-release under /etc/portables/directory so it fails

**Comment by bluca** [comment]
CI errors seem unrelated (networkd and ExecReload)

**Comment by bluca** [comment]
superseded by https://github.com/systemd/systemd/pull/14689
